src/qunitmulti.cpp

```diff
@@ -45,14 +45,15 @@ void QUnitMulti::RedistributeQEngines()
     }
 
     bitCapInt partSize = 0;
-    int devicesLeft = deviceCount;
+    int deviceId = 0;
     for (bitLenInt i = 0; i < qips.size(); i++) {
+        (dynamic_cast<QEngineOCL*>(qips[i].get()))->SetDevice(deviceIDs[deviceId]);
```

Comment:
This makes me a little nervous. Is there some other guarantee that no non-QEngineOCL engines will be added to qips?

Reply:
Yes, there is a guarantee: there is no engine type option for a QUnitMulti. It will only instantiate QEngineOCL subengines in its constructor.

Reply:
Actually, better - add a comment where the devices are acquired stating that it's a requirement that they are QEngineOCl objects.

Reply:
Done.